Try to link to non-existant utf8cpp library (ebml 1.4.6)

Debian unstable
GCC 15.3.0

utfcpp https://github.com/nemtrif/utfcpp only provides include files not shred/static library
and the build fail with:
Removing ``#target_link_libraries(ebml` PRIVATE $<BUILD_INTERFACE:utf8cpp>)` in CMakeLists.txt solve this issue.

```
FAILED: [code=1] libebml.so.5.0.0 
: && /usr/lib/ccache/x86_64-linux-gnu-g++ -fPIC -g -O2 -ffile-prefix-map=/build/libebml-dmo-1.4.6=. -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -fcf-protection -Wdate-time -D_FORTIFY_SOURCE=2 -O2 -g -DNDEBUG  -shared -Wl,-z,relro -Wl,--dependency-file=CMakeFiles/ebml.dir/link.d -Wl,-soname,libebml.so.5 -o libebml.so.5.0.0 CMakeFiles/ebml.dir/src/Debug.cpp.o CMakeFiles/ebml.dir/src/EbmlBinary.cpp.o CMakeFiles/ebml.dir/src/EbmlContexts.cpp.o CMakeFiles/ebml.dir/src/EbmlCrc32.cpp.o CMakeFiles/ebml.dir/src/EbmlDate.cpp.o CMakeFiles/ebml.dir/src/EbmlDummy.cpp.o CMakeFiles/ebml.dir/src/EbmlElement.cpp.o CMakeFiles/ebml.dir/src/EbmlFloat.cpp.o CMakeFiles/ebml.dir/src/EbmlHead.cpp.o CMakeFiles/ebml.dir/src/EbmlMaster.cpp.o CMakeFiles/ebml.dir/src/EbmlSInteger.cpp.o CMakeFiles/ebml.dir/src/EbmlStream.cpp.o CMakeFiles/ebml.dir/src/EbmlString.cpp.o CMakeFiles/ebml.dir/src/EbmlSubHead.cpp.o CMakeFiles/ebml.dir/src/EbmlUInteger.cpp.o CMakeFiles/ebml.dir/src/EbmlUnicodeString.cpp.o CMakeFiles/ebml.dir/src/EbmlVersion.cpp.o CMakeFiles/ebml.dir/src/EbmlVoid.cpp.o CMakeFiles/ebml.dir/src/IOCallback.cpp.o CMakeFiles/ebml.dir/src/MemIOCallback.cpp.o CMakeFiles/ebml.dir/src/MemReadIOCallback.cpp.o CMakeFiles/ebml.dir/src/SafeReadIOCallback.cpp.o CMakeFiles/ebml.dir/src/StdIOCallback.cpp.o  -lutf8cpp && :
/usr/bin/x86_64-linux-gnu-ld.bfd: cannot find -lutf8cpp: No such file or directory
collect2: error: ld returned 1 exit status
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -48,7 +48,7 @@ function(add_cxx_flag_if_supported)
   endforeach()
 endfunction()
 
-find_package(utf8cpp 3.2.0)
+find_package(utf8cpp)
 if(NOT utf8cpp_FOUND)
   include(FetchContent REQUIRED)
   FetchContent_Declare(
@@ -60,6 +60,11 @@ if(NOT utf8cpp_FOUND)
         OVERRIDE_FIND_PACKAGE
     )
     FetchContent_MakeAvailable(utf8cpp)
+endif()
+if(TARGET utf8cpp::utf8cpp)
+  set(libebml_utf8cpp_link_libraries utf8cpp::utf8cpp)
+else()
+  set(libebml_utf8cpp_link_libraries $<BUILD_INTERFACE:utf8cpp>)
 endif()
 
 if(DEV_MODE)
@@ -138,7 +143,7 @@ if(WIN32)
   target_compile_definitions(ebml PRIVATE _CRT_SECURE_NO_WARNINGS)
 endif()
 
-target_link_libraries(ebml PRIVATE $<BUILD_INTERFACE:utf8cpp>)
+target_link_libraries(ebml PRIVATE ${libebml_utf8cpp_link_libraries})
 
 if(CMAKE_VERSION VERSION_GREATER_EQUAL "3.20")
   if(${CMAKE_CXX_BYTE_ORDER} STREQUAL "BIG_ENDIAN")

--- a/NEWS.md
+++ b/NEWS.md
@@ -36,6 +36,7 @@
   parent projects via `add_subdirectory`.
 * Enabled building shared libraries via the usual CMake definition
   `BUILD_SHARES_LIBS` (default: off).
+* Fixed cmake rules for building with utf8cpp 4.x; fixes #344
 
 # Version 1.4.3 2022-09-30
 

